Sources/Lumina/Camera/Extensions/SessionConfigurationExtension.swift

```diff
@@ -0,0 +1,230 @@
+
+extension LuminaCamera {
+  func requestVideoPermissions() {
+    self.sessionQueue.suspend()
+    AVCaptureDevice.requestAccess(for: .video) { success in
+      if success {
+        LuminaLogger.notice(message: "successfully enabled video permissions")
+        self.sessionQueue.resume()
+        self.delegate?.cameraSetupCompleted(camera: self, result: .requiresUpdate)
+      } else {
+        LuminaLogger.warning(message: "video permissions were not allowed - video feed will not show")
+        self.delegate?.cameraSetupCompleted(camera: self, result: .videoPermissionDenied)
+      }
+    }
+  }
+
+  func requestAudioPermissions() {
+    self.sessionQueue.suspend()
+    AVCaptureDevice.requestAccess(for: AVMediaType.audio) { success in
+      if success {
+        LuminaLogger.notice(message: "successfully enabled audio permissions")
+        self.sessionQueue.resume()
+        self.delegate?.cameraSetupCompleted(camera: self, result: .requiresUpdate)
+      } else {
+        LuminaLogger.warning(message: "audio permissions were not allowed - audio feed will not be present")
+        self.delegate?.cameraSetupCompleted(camera: self, result: .audioPermissionDenied)
+      }
+    }
+  }
+
+  func updateOutputVideoOrientation(_ orientation: AVCaptureVideoOrientation) {
+    self.videoBufferQueue.async {
+      for output in self.session.outputs {
+        guard let connection = output.connection(with: AVMediaType.video) else {
+          continue
+        }
+        if connection.isVideoOrientationSupported {
+          connection.videoOrientation = orientation
+        }
+      }
+    }
+  }
+
+  func restartVideo() {
+    LuminaLogger.notice(message: "restarting video feed")
+    if self.session.isRunning {
+      self.stop()
+      updateVideo { result in
+        if result == .videoSuccess {
+          self.start()
+        } else {
+          self.delegate?.cameraSetupCompleted(camera: self, result: result)
+        }
+      }
+    }
+  }
+
+  func updateAudio(_ completion: @escaping (_ result: CameraSetupResult) -> Void) {
+    self.sessionQueue.async {
+      self.purgeAudioDevices()
+      switch AVCaptureDevice.authorizationStatus(for: AVMediaType.audio) {
+        case .authorized:
+          guard let audioInput = self.getNewAudioInputDevice() else {
+            return completion(CameraSetupResult.invalidAudioInput)
+          }
+          guard self.session.canAddInput(audioInput) else {
+            return completion(CameraSetupResult.invalidAudioInput)
+          }
+          self.audioInput = audioInput
+          self.session.addInput(audioInput)
+          return completion(CameraSetupResult.audioSuccess)
+        case .denied:
+          return completion(CameraSetupResult.audioPermissionDenied)
+        case .notDetermined:
+          return completion(CameraSetupResult.audioRequiresAuthorization)
+        case .restricted:
+          return completion(CameraSetupResult.audioPermissionRestricted)
+        @unknown default:
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dokun1/Lumina/blob/f3c4aa8d5c66dd9d2660cdba8e658b8902c9752f/Sources/Lumina/Camera/Extensions/SessionConfigurationExtension.swift#L88"></span>Case statements should vertically align with their enclosing switch statement.<br />
<code>switch_case_alignment</code> <code>SessionConfigurationExtension.swift:88</code></td>
    </tr>
  </tbody>
</table>